include/rocksdb/write_batch_base.h

```diff
@@ -55,11 +55,19 @@ class WriteBatchBase {
   virtual Status Merge(ColumnFamilyHandle* column_family, const Slice& key,
                        const Slice& ts, const Slice& value) = 0;
 
-  // variant that takes SliceParts
   virtual Status Merge(ColumnFamilyHandle* column_family, const SliceParts& key,
                        const SliceParts& value);
   virtual Status Merge(const SliceParts& key, const SliceParts& value);
 
+  // Eager variants
+  struct EagerMergeMode {};
+  static constexpr EagerMergeMode kEagerMerge{};
+
+  virtual Status Merge(EagerMergeMode, ColumnFamilyHandle* column_family,
```

Comment:
Not a huge fan of this way of overloading. In particular, if we have eventually have n ways of varying Merge(), we might have 2^n overloads.

Reply:
Tag dispatch is a tried and true technique but you're right that it can result in a combinatorial explosion. On a practical note, with our warning settings, adding an overload like this requires putting a using declaration in each WriteBatch::Handler. So in case I get back to this, I'll probably switch to a separate EagerMerge API.